nilearn/image/resampling.py

```diff
@@ -455,8 +455,13 @@ def resample_img(img, target_affine=None, target_shape=None,
     target_shape = tuple(target_shape)
 
     if interpolation == 'continuous' and data.dtype.kind == 'i':
-        resampled_data_dtype = np.dtype(
-            data.dtype.name.replace('int', 'float'))
+        # cast unsupported data types to closest support dtype
+        aux = data.dtype.name.replace('int', 'float')
+        aux = aux.replace("ufloat", "float").replace("floatc", "float")
```

Comment:
If you promise to rename the `aux` variable, I would say push the change into master :-)